dts/bindings/base/pm.yaml

```diff
@@ -16,6 +16,11 @@ properties:
       Wake up capable devices are disabled (interruptions will not wake up
       the system) by default but they can be enabled at runtime if necessary.
 
+  wakeup-source-pin:
+    type: uint8-array
+    description: |
+      Property to identify that which pins can be used as wake up source.
+
```

Comment:
Not sure this belongs in the generic, common `pm.yaml` file? Should it be added to either the LPC GPIO binding directly or perhaps in either `gpio-controller.yaml` or a new `gpio-controller-pm.yaml`?

The `wakeup-source` property indicates that this particular DTS device node can be used for waking up the system/CPU. `wakeup-source-pin` sounds more specific to DTS device nodes representing multiple pins, such as GPIO controllers - and not all GPIO controllers supports setting which pin is capable of waking up the system/CPU.

Reply:
I agree - it should be possible to factor out to its own binding, but it would be nice to simply have a bool to indicate that the controller is a wakeup source and then to have further optional specialisation as an array of specific pins for gpio controllers that can support individual pins.

Reply:
Agree. `wakeup-source` is used to tell the pm subsystem that a device is used to wake up the system and it should not be suspended when the SoC is sleeping.  Devices themselves can use it for other purposes but that sort of specialization should not be done inside the pm subsystem.